src/App.jsx

```diff
@@ -1,28 +1,85 @@
 import React from 'react';
+
 import './App.scss';
 import './styles/general.scss';
+
+import getData from './api/api';
+
 import { TodoList } from './components/TodoList';
 import { CurrentUser } from './components/CurrentUser';
 
+const url = 'todos';
+
 class App extends React.Component {
   state = {
     todos: [],
     selectedUserId: 0,
+    titleFilter: '',
+    todoStatus: 'all',
   };
 
+  componentDidMount() {
+    getData(url)
+      .then((todos) => {
+        this.setState({ todos });
+      });
+  }
+
+  selectUser = (userId) => {
+    if (userId !== this.state.selectedUserId) {
+      this.setState({ selectedUserId: userId });
+    }
+  }
+
+  clearUser = () => {
+    this.setState({ selectedUserId: 0 });
+  }
+
+  inputHandler = (event) => {
+    this.setState({ titleFilter: event.target.value });
+  }
+
+  selectHandler = (event) => {
+    this.setState({ todoStatus: event.target.value });
+  }
+
   render() {
-    const { todos, selectedUserId } = this.state;
+    const { todos, selectedUserId, titleFilter, todoStatus } = this.state;
+
+    const formatedList = todos
+      .filter(todo => !Object.values(todo).includes(null))
```

Comment:
Not the best way to filter. If no filter value is specified you should show all todos.
To avoid errors you can filter out todos with `null` titles when any filter value has been specified.